### PR TITLE
fix up serialization and deserialization of events

### PR DESCRIPTION
**Pull request by rbren**
I noticed a few problems while working on https://github.com/OpenDevin/OpenDevin/pull/1810

Breaking the fixes out into this PR

### CODE REVIEW

File: tests/unit/test_action_serialization.py [hunk 1]
```diff
@@ -14,31 +14,52 @@
...
     assert isinstance(
         action_instance, Action
     ), 'The action instance should be an instance of Action.'
     assert isinstance(
         action_instance, cls
     ), f'The action instance should be an instance of {cls.__name__}.'
+    print('got action_instance', action_instance)
+    print('id', action_instance.id)
+    print('source', action_instance.source)
+    print('timestamp', action_instance.timestamp)
```
Comment (enyst) [inline tests/unit/test_action_serialization.py:34]:
```suggestion
```
😄

Comment (rbren) [reply tests/unit/test_action_serialization.py]: 🤦

**Review by enyst — APPROVED** [review]

File: opendevin/controller/state/state.py [hunk 2]
```diff
@@ -27,6 +30,19 @@ class State:
     error: str | None = None
     agent_state: AgentState = AgentState.LOADING
 
+    def save_to_session(self, sid: str):
+        fs = get_file_store()
```
Comment (isavita) [inline opendevin/controller/state/state.py:34]: What about adding `try` block
```python
def save_to_session(self, sid: str):
    try:
        fs = get_file_store()
        pickled = pickle.dumps(self)
        encoded = base64.b64encode(pickled).decode('utf-8')
        fs.write(f'sessions/{sid}/agent_state.pkl', encoded)
    except Exception as e:
        print(f"Error saving session: {e}")
```

File: opendevin/controller/state/state.py [hunk 3]
```diff
@@ -27,6 +30,19 @@ class State:
...
+        pickled = pickle.dumps(self)
+        encoded = base64.b64encode(pickled).decode('utf-8')
+        fs.write(f'sessions/{sid}/agent_state.pkl', encoded)
+
+    @staticmethod
+    def restore_from_session(sid: str) -> 'State':
+        fs = get_file_store()
```
Comment (isavita) [inline opendevin/controller/state/state.py:41]: Probably and here same `try` with on `Exception return some `None`.

File: opendevin/controller/state/state.py (hunk 2 above)
Comment (rbren) [reply opendevin/controller/state/state.py:35]: Thanks! I think we should continue to pass the error through, but I logged it at least

**Pull request merged**